src/test/scala/it/pagopa/interop/purposeprocess/PurposeApiServiceSpec.scala

```diff
@@ -32,6 +32,414 @@ class PurposeApiServiceSpec extends AnyWordSpecLike with SpecHelper with Scalate
+            expectedApprovalDate = None,
+            dailyCalls = 50,
+            riskAnalysis = None
+          )
+        ),
+        suspendedByConsumer = None,
+        suspendedByProducer = None,
+        title = "title",
+        description = "description",
+        riskAnalysisForm = Some(SpecData.validManagementRiskAnalysis),
+        createdAt = SpecData.timestamp,
+        updatedAt = None
+      )
+
+      val purposeCloned = PurposeManagementDependency.Purpose(
+        id = UUID.randomUUID(),
+        eserviceId = eServiceId,
+        consumerId = consumerId,
+        versions = Seq(SpecData.purposeVersion),
+        suspendedByConsumer = None,
+        suspendedByProducer = None,
+        title = "title - clone",
+        description = "description",
+        riskAnalysisForm = Some(SpecData.validManagementRiskAnalysis),
+        createdAt = SpecData.timestamp,
+        updatedAt = None
+      )
+
+      (mockPurposeManagementService
+        .getPurpose(_: UUID)(_: Seq[(String, String)]))
+        .expects(purposeId, context)
+        .once()
+        .returns(Future.successful(purposeToClone))
+
+      (mockPurposeManagementService
+        .createPurpose(_: PurposeManagementDependency.PurposeSeed)(_: Seq[(String, String)]))
+        .expects(*, context)
+        .once()
+        .returns(Future.successful(purposeCloned))
+
+      (mockPurposeManagementService
+        .createPurposeVersion(_: UUID, _: PurposeManagementDependency.PurposeVersionSeed)(_: Seq[(String, String)]))
+        .expects(purposeCloned.id, PurposeManagementDependency.PurposeVersionSeed(50, None), context)
+        .once()
+        .returns(Future.successful(SpecData.purposeVersion))
+
+      Get() ~> service.clonePurpose(purposeId.toString) ~> check {
+        status shouldEqual StatusCodes.OK
+      }
+    }
+    "succeed with multiple versions no WAITING FOR APPROVAL" in {
+      val eServiceId = UUID.randomUUID()
+      val consumerId = UUID.randomUUID()
+      val purposeId  = UUID.randomUUID()
+
+      implicit val context: Seq[(String, String)] =
+        Seq("bearer" -> bearerToken, USER_ROLES -> "admin", ORGANIZATION_ID_CLAIM -> consumerId.toString)
+
+      val purposeToClone = PurposeManagementDependency.Purpose(
+        id = purposeId,
+        eserviceId = eServiceId,
+        consumerId = consumerId,
+        versions = Seq(
+          PurposeManagementDependency.PurposeVersion(
+            id = UUID.randomUUID(),
+            state = PurposeManagementDependency.PurposeVersionState.ACTIVE,
+            createdAt = OffsetDateTime.of(2022, 11, 30, 11, 22, 33, 44, ZoneOffset.UTC),
+            updatedAt = None,
+            firstActivationAt = None,
+            expectedApprovalDate = None,
+            dailyCalls = 50,
+            riskAnalysis = None
+          ),
+          PurposeManagementDependency.PurposeVersion(
+            id = UUID.randomUUID(),
+            state = PurposeManagementDependency.PurposeVersionState.SUSPENDED,
```

Comment:
Just to have a more realistic case (only one version in a purpose can be ACTIVE or SUSPENDED in a given time)
```suggestion
            state = PurposeManagementDependency.PurposeVersionState.ARCHIVED,
```